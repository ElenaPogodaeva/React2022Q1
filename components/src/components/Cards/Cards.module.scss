.cards {
  width: 100%;
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  padding: 20px 0;
  background: #f0f0f0;
}
