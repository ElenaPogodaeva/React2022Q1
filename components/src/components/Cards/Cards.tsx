import React from 'react';
import Card from '../Card/Card';

import style from './Cards.module.scss';

type CardProps = {
  id: string;
  title: string;
  author: string;
  url: string;
  date: string;
  views: string;
  likes: string;
  dislikes: string;
  comments: string;
};

type CardsProps = {
  cards: CardProps[];
};

type CardsState = {
  state: Record<string, never>;
};

class Cards extends React.Component<CardsProps, CardsState> {
  render() {
    if (!this.props.cards.length) return null;

    return (
      <div className={style.cards} data-testid="cards">
        {this.props.cards &&
          this.props.cards.map((card) => <Card key={card.id} {...card} />)}
      </div>
    );
  }
}

export default Cards;
