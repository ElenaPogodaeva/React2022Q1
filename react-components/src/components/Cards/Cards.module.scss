.cards {
  width: 100%;
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  padding: 10px 0;
}
