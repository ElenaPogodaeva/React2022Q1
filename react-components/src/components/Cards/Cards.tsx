import React from 'react';
import Card from '../Card/Card';
import { Image, ImageInfo, ImageSize, SearchInfoParams } from '../../types/types';
import style from './Cards.module.scss';
import ImageDetail from '../ImageDetail/ImageDetail';
import Spinner from '../Spinner/Spinner';
import { flickr } from '../../common/flickr';

type CardsProps = {
  cards: Image[];
};

type CardsState = {
  currentImageId: string;
  imageInfo: ImageInfo | null;
  imageUrl: string;
  viewerIsOpen: boolean;
  isLoading: boolean;
};

class Cards extends React.Component<CardsProps, CardsState> {
  constructor(props: CardsProps) {
    super(props);
    this.state = {
      currentImageId: '',
      imageInfo: null,
      imageUrl: '',
      viewerIsOpen: false,
      isLoading: true,
    };
    this.handleClick = this.handleClick.bind(this);
    this.handleClose = this.handleClose.bind(this);
  }

  async fetchImageInfo() {
    const currentId = this.state.currentImageId;

    const params: SearchInfoParams = {
      photo_id: currentId,
    };

    try {
      const fetchedImageInfo = await flickr('photos.getInfo', params);
      this.setState({
        imageInfo: fetchedImageInfo.photo,
      });
    } catch (err) {
      console.error(err);
    }
  }

  async fetchImageSizes() {
    const currentId = this.state.currentImageId;

    const params: SearchInfoParams = {
      photo_id: currentId,
    };

    try {
      const fetchedImageSizes = await flickr('photos.getSizes', params);
      const sizes = fetchedImageSizes.sizes.size;

      this.setState({
        imageUrl: sizes
          .reverse()
          .filter((s: ImageSize) => s.label === 'Small' || s.label === 'Large')[0].source,
      });
    } catch (err) {
      console.error(err);
    }
  }

  async handleClick(id: string) {
    await this.setState({
      currentImageId: id,
      viewerIsOpen: true,
      isLoading: true,
    });
    await this.fetchImageInfo();
    await this.fetchImageSizes();
    await this.setState({
      isLoading: false,
    });
  }

  handleClose() {
    this.setState({
      viewerIsOpen: false,
    });
  }

  render() {
    if (!this.props.cards.length) return null;

    return (
      <>
        <div className={style.cards} data-testid="cards">
          {this.props.cards &&
            this.props.cards.map((card) => (
              <Card key={card.id} card={card} handleClick={this.handleClick} />
            ))}
        </div>
        {this.state.viewerIsOpen ? (
          this.state.isLoading ? (
            <Spinner />
          ) : (
            <ImageDetail
              imageInfo={this.state.imageInfo as ImageInfo}
              imageUrl={this.state.imageUrl}
              handleClose={this.handleClose}
            />
          )
        ) : null}
      </>
    );
  }
}

export default Cards;
