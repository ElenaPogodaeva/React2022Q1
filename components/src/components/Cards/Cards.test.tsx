import { render } from '@testing-library/react';
import Cards from './Cards';

const propsValues = [
  {
    id: '1',
    title: 'Portrait of a cat looking at the camera',
    author: 'Lloyd Henneman',
    url: '',
    date: 'March 26th, 2022',
    views: '1000',
    likes: '500',
    dislikes: '200',
    comments: '100',
  },
  {
    id: '2',
    title: 'Gray cat',
    author: 'Max Baskakov',
    url: '',
    date: 'March 26th, 2022',
    views: '1000',
    likes: '500',
    dislikes: '200',
    comments: '100',
  },
  {
    id: '3',
    title: 'Stretching in the morning',
    author: 'Timo Volz',
    url: '',
    date: 'March 26th, 2022',
    views: '1000',
    likes: '500',
    dislikes: '200',
    comments: '100',
  },
  {
    id: '4',
    title: 'Gray cat with flowers',
    author: 'Milada Vigerova',
    url: '',
    date: 'March 28th, 2022',
    views: '1000',
    likes: '500',
    dislikes: '200',
    comments: '100',
  },
  {
    id: '5',
    title: 'Kitten',
    author: 'Alvan Nee',
    url: '',
    date: 'March 28th, 2022',
    views: '1000',
    likes: '500',
    dislikes: '200',
    comments: '100',
  },
  {
    id: '6',
    title: 'Two kittens sitting in a basket',
    author: 'Amy Baugess',
    url: '',
    date: 'March 28th, 2022',
    views: '1000',
    likes: '500',
    dislikes: '200',
    comments: '100',
  },
  {
    id: '7',
    title: 'Red cat lying on a wooden floor',
    author: 'Michael Sum',
    url: '',
    date: 'March 28th, 2022',
    views: '1000',
    likes: '500',
    dislikes: '200',
    comments: '100',
  },
  {
    id: '8',
    title: 'Yawn cat',
    author: 'Loan',
    url: '',
    date: 'March 28th, 2022',
    views: '1000',
    likes: '500',
    dislikes: '200',
    comments: '100',
  },
];

describe('Cards component', () => {
  it('Cards renders', () => {
    const { getByTestId } = render(<Cards cards={propsValues} />);
    const cards = getByTestId('cards');
    expect(cards).toBeDefined();
  });
  it('Cards renders without data', () => {
    const { queryByTestId } = render(<Cards cards={[]} />);
    const cards = queryByTestId('cards');
    expect(cards).toBeNull();
  });
  it('Cards snapshot', () => {
    const cards = render(<Cards cards={propsValues} />);
    expect(cards).toMatchSnapshot();
  });
  it('Cards empty snapshot', () => {
    const cards = render(<Cards cards={[]} />);
    expect(cards).toMatchSnapshot();
  });
});
