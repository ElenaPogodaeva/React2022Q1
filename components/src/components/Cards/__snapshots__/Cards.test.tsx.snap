// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards component Cards empty snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Cards component Cards snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cards"
        data-testid="cards"
      >
        <div
          class="card"
          data-testid="card"
        >
          <img
            class="cardImg"
            src="1.jpg"
          />
          <div
            class="cardBody"
          >
            <p
              class="cardTitle"
            >
              Portrait of a cat looking at the camera
            </p>
            <p
              class="cardOwner"
            >
              By 
              Lloyd Henneman
            </p>
            <p
              class="cardDate"
            >
              March 26th, 2022
            </p>
            <div
              class="cardStatistics"
            >
              <div
                class="statisticsItem"
              >
                <div
                  class="statisticsImg statisticsViews"
                />
                1000
              </div>
              <div
                class="statisticsItem"
              >
                <div
                  class="statisticsImg statisticsLikes"
                />
                500
              </div>
              <div
                class="statisticsItem"
              >
                <div
                  class="statisticsImg statisticsComments"
                />
                100
              </div>
            </div>
          </div>
        </div>
        <div
          class="card"
          data-testid="card"
        >
          <img
            class="cardImg"
            src="2.jpg"
          />
          <div
            class="cardBody"
          >
            <p
              class="cardTitle"
            >
              Gray cat
            </p>
            <p
              class="cardOwner"
            >
              By 
              Max Baskakov
            </p>
            <p
              class="cardDate"
            >
              March 26th, 2022
            </p>
            <div
              class="cardStatistics"
            >
              <div
                class="statisticsItem"
              >
                <div
                  class="statisticsImg statisticsViews"
                />
                1000
              </div>
              <div
                class="statisticsItem"
              >
                <div
                  class="statisticsImg statisticsLikes"
                />
                500
              </div>
              <div
                class="statisticsItem"
              >
                <div
                  class="statisticsImg statisticsComments"
                />
                100
              </div>
            </div>
          </div>
        </div>
        <div
          class="card"
          data-testid="card"
        >
          <img
            class="cardImg"
            src="3.jpg"
          />
          <div
            class="cardBody"
          >
            <p
              class="cardTitle"
            >
              Stretching in the morning
            </p>
            <p
              class="cardOwner"
            >
              By 
              Timo Volz
            </p>
            <p
              class="cardDate"
            >
              March 26th, 2022
            </p>
            <div
              class="cardStatistics"
            >
              <div
                class="statisticsItem"
              >
                <div
                  class="statisticsImg statisticsViews"
                />
                1000
              </div>
              <div
                class="statisticsItem"
              >
                <div
                  class="statisticsImg statisticsLikes"
                />
                500
              </div>
              <div
                class="statisticsItem"
              >
                <div
                  class="statisticsImg statisticsComments"
                />
                100
              </div>
            </div>
          </div>
        </div>
        <div
          class="card"
          data-testid="card"
        >
          <img
            class="cardImg"
            src="4.jpg"
          />
          <div
            class="cardBody"
          >
            <p
              class="cardTitle"
            >
              Gray cat with flowers
            </p>
            <p
              class="cardOwner"
            >
              By 
              Milada Vigerova
            </p>
            <p
              class="cardDate"
            >
              March 28th, 2022
            </p>
            <div
              class="cardStatistics"
            >
              <div
                class="statisticsItem"
              >
                <div
                  class="statisticsImg statisticsViews"
                />
                1000
              </div>
              <div
                class="statisticsItem"
              >
                <div
                  class="statisticsImg statisticsLikes"
                />
                500
              </div>
              <div
                class="statisticsItem"
              >
                <div
                  class="statisticsImg statisticsComments"
                />
                100
              </div>
            </div>
          </div>
        </div>
        <div
          class="card"
          data-testid="card"
        >
          <img
            class="cardImg"
            src="5.jpg"
          />
          <div
            class="cardBody"
          >
            <p
              class="cardTitle"
            >
              Kitten
            </p>
            <p
              class="cardOwner"
            >
              By 
              Alvan Nee
            </p>
            <p
              class="cardDate"
            >
              March 28th, 2022
            </p>
            <div
              class="cardStatistics"
            >
              <div
                class="statisticsItem"
              >
                <div
                  class="statisticsImg statisticsViews"
                />
                1000
              </div>
              <div
                class="statisticsItem"
              >
                <div
                  class="statisticsImg statisticsLikes"
                />
                500
              </div>
              <div
                class="statisticsItem"
              >
                <div
                  class="statisticsImg statisticsComments"
                />
                100
              </div>
            </div>
          </div>
        </div>
        <div
          class="card"
          data-testid="card"
        >
          <img
            class="cardImg"
            src="6.jpg"
          />
          <div
            class="cardBody"
          >
            <p
              class="cardTitle"
            >
              Two kittens sitting in a basket
            </p>
            <p
              class="cardOwner"
            >
              By 
              Amy Baugess
            </p>
            <p
              class="cardDate"
            >
              March 28th, 2022
            </p>
            <div
              class="cardStatistics"
            >
              <div
                class="statisticsItem"
              >
                <div
                  class="statisticsImg statisticsViews"
                />
                1000
              </div>
              <div
                class="statisticsItem"
              >
                <div
                  class="statisticsImg statisticsLikes"
                />
                500
              </div>
              <div
                class="statisticsItem"
              >
                <div
                  class="statisticsImg statisticsComments"
                />
                100
              </div>
            </div>
          </div>
        </div>
        <div
          class="card"
          data-testid="card"
        >
          <img
            class="cardImg"
            src="7.jpg"
          />
          <div
            class="cardBody"
          >
            <p
              class="cardTitle"
            >
              Red cat lying on a wooden floor
            </p>
            <p
              class="cardOwner"
            >
              By 
              Michael Sum
            </p>
            <p
              class="cardDate"
            >
              March 28th, 2022
            </p>
            <div
              class="cardStatistics"
            >
              <div
                class="statisticsItem"
              >
                <div
                  class="statisticsImg statisticsViews"
                />
                1000
              </div>
              <div
                class="statisticsItem"
              >
                <div
                  class="statisticsImg statisticsLikes"
                />
                500
              </div>
              <div
                class="statisticsItem"
              >
                <div
                  class="statisticsImg statisticsComments"
                />
                100
              </div>
            </div>
          </div>
        </div>
        <div
          class="card"
          data-testid="card"
        >
          <img
            class="cardImg"
            src="8.jpg"
          />
          <div
            class="cardBody"
          >
            <p
              class="cardTitle"
            >
              Yawn cat
            </p>
            <p
              class="cardOwner"
            >
              By 
              Loan
            </p>
            <p
              class="cardDate"
            >
              March 28th, 2022
            </p>
            <div
              class="cardStatistics"
            >
              <div
                class="statisticsItem"
              >
                <div
                  class="statisticsImg statisticsViews"
                />
                1000
              </div>
              <div
                class="statisticsItem"
              >
                <div
                  class="statisticsImg statisticsLikes"
                />
                500
              </div>
              <div
                class="statisticsItem"
              >
                <div
                  class="statisticsImg statisticsComments"
                />
                100
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="cards"
      data-testid="cards"
    >
      <div
        class="card"
        data-testid="card"
      >
        <img
          class="cardImg"
          src="1.jpg"
        />
        <div
          class="cardBody"
        >
          <p
            class="cardTitle"
          >
            Portrait of a cat looking at the camera
          </p>
          <p
            class="cardOwner"
          >
            By 
            Lloyd Henneman
          </p>
          <p
            class="cardDate"
          >
            March 26th, 2022
          </p>
          <div
            class="cardStatistics"
          >
            <div
              class="statisticsItem"
            >
              <div
                class="statisticsImg statisticsViews"
              />
              1000
            </div>
            <div
              class="statisticsItem"
            >
              <div
                class="statisticsImg statisticsLikes"
              />
              500
            </div>
            <div
              class="statisticsItem"
            >
              <div
                class="statisticsImg statisticsComments"
              />
              100
            </div>
          </div>
        </div>
      </div>
      <div
        class="card"
        data-testid="card"
      >
        <img
          class="cardImg"
          src="2.jpg"
        />
        <div
          class="cardBody"
        >
          <p
            class="cardTitle"
          >
            Gray cat
          </p>
          <p
            class="cardOwner"
          >
            By 
            Max Baskakov
          </p>
          <p
            class="cardDate"
          >
            March 26th, 2022
          </p>
          <div
            class="cardStatistics"
          >
            <div
              class="statisticsItem"
            >
              <div
                class="statisticsImg statisticsViews"
              />
              1000
            </div>
            <div
              class="statisticsItem"
            >
              <div
                class="statisticsImg statisticsLikes"
              />
              500
            </div>
            <div
              class="statisticsItem"
            >
              <div
                class="statisticsImg statisticsComments"
              />
              100
            </div>
          </div>
        </div>
      </div>
      <div
        class="card"
        data-testid="card"
      >
        <img
          class="cardImg"
          src="3.jpg"
        />
        <div
          class="cardBody"
        >
          <p
            class="cardTitle"
          >
            Stretching in the morning
          </p>
          <p
            class="cardOwner"
          >
            By 
            Timo Volz
          </p>
          <p
            class="cardDate"
          >
            March 26th, 2022
          </p>
          <div
            class="cardStatistics"
          >
            <div
              class="statisticsItem"
            >
              <div
                class="statisticsImg statisticsViews"
              />
              1000
            </div>
            <div
              class="statisticsItem"
            >
              <div
                class="statisticsImg statisticsLikes"
              />
              500
            </div>
            <div
              class="statisticsItem"
            >
              <div
                class="statisticsImg statisticsComments"
              />
              100
            </div>
          </div>
        </div>
      </div>
      <div
        class="card"
        data-testid="card"
      >
        <img
          class="cardImg"
          src="4.jpg"
        />
        <div
          class="cardBody"
        >
          <p
            class="cardTitle"
          >
            Gray cat with flowers
          </p>
          <p
            class="cardOwner"
          >
            By 
            Milada Vigerova
          </p>
          <p
            class="cardDate"
          >
            March 28th, 2022
          </p>
          <div
            class="cardStatistics"
          >
            <div
              class="statisticsItem"
            >
              <div
                class="statisticsImg statisticsViews"
              />
              1000
            </div>
            <div
              class="statisticsItem"
            >
              <div
                class="statisticsImg statisticsLikes"
              />
              500
            </div>
            <div
              class="statisticsItem"
            >
              <div
                class="statisticsImg statisticsComments"
              />
              100
            </div>
          </div>
        </div>
      </div>
      <div
        class="card"
        data-testid="card"
      >
        <img
          class="cardImg"
          src="5.jpg"
        />
        <div
          class="cardBody"
        >
          <p
            class="cardTitle"
          >
            Kitten
          </p>
          <p
            class="cardOwner"
          >
            By 
            Alvan Nee
          </p>
          <p
            class="cardDate"
          >
            March 28th, 2022
          </p>
          <div
            class="cardStatistics"
          >
            <div
              class="statisticsItem"
            >
              <div
                class="statisticsImg statisticsViews"
              />
              1000
            </div>
            <div
              class="statisticsItem"
            >
              <div
                class="statisticsImg statisticsLikes"
              />
              500
            </div>
            <div
              class="statisticsItem"
            >
              <div
                class="statisticsImg statisticsComments"
              />
              100
            </div>
          </div>
        </div>
      </div>
      <div
        class="card"
        data-testid="card"
      >
        <img
          class="cardImg"
          src="6.jpg"
        />
        <div
          class="cardBody"
        >
          <p
            class="cardTitle"
          >
            Two kittens sitting in a basket
          </p>
          <p
            class="cardOwner"
          >
            By 
            Amy Baugess
          </p>
          <p
            class="cardDate"
          >
            March 28th, 2022
          </p>
          <div
            class="cardStatistics"
          >
            <div
              class="statisticsItem"
            >
              <div
                class="statisticsImg statisticsViews"
              />
              1000
            </div>
            <div
              class="statisticsItem"
            >
              <div
                class="statisticsImg statisticsLikes"
              />
              500
            </div>
            <div
              class="statisticsItem"
            >
              <div
                class="statisticsImg statisticsComments"
              />
              100
            </div>
          </div>
        </div>
      </div>
      <div
        class="card"
        data-testid="card"
      >
        <img
          class="cardImg"
          src="7.jpg"
        />
        <div
          class="cardBody"
        >
          <p
            class="cardTitle"
          >
            Red cat lying on a wooden floor
          </p>
          <p
            class="cardOwner"
          >
            By 
            Michael Sum
          </p>
          <p
            class="cardDate"
          >
            March 28th, 2022
          </p>
          <div
            class="cardStatistics"
          >
            <div
              class="statisticsItem"
            >
              <div
                class="statisticsImg statisticsViews"
              />
              1000
            </div>
            <div
              class="statisticsItem"
            >
              <div
                class="statisticsImg statisticsLikes"
              />
              500
            </div>
            <div
              class="statisticsItem"
            >
              <div
                class="statisticsImg statisticsComments"
              />
              100
            </div>
          </div>
        </div>
      </div>
      <div
        class="card"
        data-testid="card"
      >
        <img
          class="cardImg"
          src="8.jpg"
        />
        <div
          class="cardBody"
        >
          <p
            class="cardTitle"
          >
            Yawn cat
          </p>
          <p
            class="cardOwner"
          >
            By 
            Loan
          </p>
          <p
            class="cardDate"
          >
            March 28th, 2022
          </p>
          <div
            class="cardStatistics"
          >
            <div
              class="statisticsItem"
            >
              <div
                class="statisticsImg statisticsViews"
              />
              1000
            </div>
            <div
              class="statisticsItem"
            >
              <div
                class="statisticsImg statisticsLikes"
              />
              500
            </div>
            <div
              class="statisticsItem"
            >
              <div
                class="statisticsImg statisticsComments"
              />
              100
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
